src/control/dmg/main.go

```diff
@@ -83,12 +91,20 @@ func dmgMain() error {
 		return err
 	}
 
-	// TODO: implement configuration file parsing
-	if opts.ConfigPath != "" {
-		err = errors.New("config-path option not implemented")
-		log.Errorf(err.Error())
+	// Load the configuration file using the supplied path or the default path if none provided.
+	config, err = client.ProcessConfigFile(opts.ConfigPath)
+	if err != nil {
+		log.Errorf("An unrecoverable error occurred while processing the configuration file: %s", err)
+		return err
+	}
+
+	// Override configuration with any commandline values given
+	err = applyCmdLineOverrides(&config, opts.Hostlist)
```

Comment:
again this should take opts, but if it's only the single override why don't we apply the logic above as it was before

Reply:
Having applyCmdLineOverrides seemed to be the right format for handling this sort of thing, with an eye towards adding more as DMG expands.  I am ok with it either way.  If you want the applyCmdLineOverrides removed because there's only one override, I can do that.  What would you like at this time?